indexing-service/src/main/java/org/apache/druid/indexing/common/actions/CheckPointDataSourceMetadataAction.java

```diff
@@ -114,4 +137,29 @@ public String toString()
            ", checkpointMetadata=" + checkpointMetadata +
            '}';
   }
+
+  @Override
+  public boolean equals(Object o)
```

Comment:
Since you have an override for `equals`, you'll need to also override `hashCode`

FYI, the [SonarLint intellij plugin](https://plugins.jetbrains.com/plugin/7973-sonarlint) will give you a warning if you don't override both.

While typing this, I just noticed LGTM flagged the same issue.